internal/mode/static/state/graph/gatewayclass.go

```diff
@@ -77,20 +79,33 @@ func buildGatewayClass(
 
 func validateGatewayClass(
 	gc *v1.GatewayClass,
+	npCfg *ngfAPI.NginxProxy,
 	crdVersions map[types.NamespacedName]*metav1.PartialObjectMetadata,
 ) ([]conditions.Condition, bool) {
 	var conds []conditions.Condition
 
-	valid := true
-
 	if gc.Spec.ParametersRef != nil {
+		var err error
 		path := field.NewPath("spec").Child("parametersRef")
-		err := field.Forbidden(path, "parametersRef is not supported")
-		conds = append(conds, staticConds.NewGatewayClassInvalidParameters(err.Error()))
-		valid = false
+		if _, ok := supportedParamKinds[string(gc.Spec.ParametersRef.Kind)]; !ok {
```

Comment:
I think we should validate some of the fields of NginxProxy here.

Specifically, all strings that go into NGINX config.

This can be a fall back against any bugs or misconfiguration in CRD validation.

Such validation can prevent config injection (vulnerability) by preventing string values escaping surrounding "...". 

At the same time, not fully validating values is OK. In rare case CRD validation doesn't work, NGINX will fail to reload and errors will be reported on Gateway (not programmed). But at least, it will not be possible to inject malicious config.

Note, for such validation, we have Validator struct https://github.com/nginxinc/nginx-gateway-fabric/blob/d3de0c68f088e3efc75be3cbe0960e6a3a1aeeca/internal/mode/static/state/validation/validator.go#L8

Reply:
Do you mean just duplicate the same regex checking that we do in the CRD? What additional validation should we do?

Reply:
> Do you mean just duplicate the same regex checking that we do in the CRD?

At minimum, I think we need to add validation to prevent config injection. Which is disallowing unescaped `"` and `\` at the end. 

At maximum, we can do the same validation as we do in CRDs, so that not only config inject, but invalid values will be rejected. However, not fully sure about it, because of the duplicated efforts.

Reply:
If we are going to be validating all strings to prevent injection, it's not any more work to validate the string against the regex in the CRD, and it guarantees that the value is valid.

Reply:
Created a GenericValidator since this validation doesn't fall into the existing HTTPFieldsValidator by name. Let me know if this approach makes sense or if I'm wildly off and there's something much simpler.

Reply:
> If we are going to be validating all strings to prevent injection, it's not any more work to validate the string against the regex in the CRD, and it guarantees that the value is valid.


I wonder if we just repeat the same validation in this case.

> Created a GenericValidator since this validation doesn't fall into the existing HTTPFieldsValidator by name. Let me know if this approach makes sense or if I'm wildly off and there's something much simpler.

the approach with GenericValidator lgmt. 

But for `serviceName`, `endpoint`, `interval` and `key` -  shall we just do them same CRD validation then?

Reply:
> But for serviceName, endpoint, interval and key - shall we just do them same CRD validation then?
Ah yeah, I did the worst case scenario, but I should just do the same regex.